src/canvas/Document.tsx

```diff
@@ -132,18 +137,68 @@ export class Document implements IDocument {
         } 
         return this._sizeRatio;        
     };
-    
+
+    getAllowanceSize = (type: PatternPathType): number => {
+        if (!this._allowanceSizes) {
+            throw new Error("Allowance sizes were not set yet.");
+        }
+        const allowance = this._allowanceSizes.get(type);
+        if (allowance === undefined) {
+            throw new Error("Could not find allowance for this path type.");
+        }
+        return allowance;
+    };
+
+    /**
+     * Sets the allowance sizes to 0 inches for folds, the input edgeAllowance 
+     * for edges and the input seam allowance for seams. If parameters are
+     * omitted, the edge allowance will be set to a default of 5/8 inch 
+     * and the seam allowance to a default of 5/8 inch. The sizes are stored in
+     * pixels in the allowanceSizes map data field.
+     * 
+     * Warning: every time this method is called the old data is erased.
+     *
+     * @param edgeAllowance 
+     * @param seamAllowance 
+     */
+    setAllowanceSizes = (edgeAllowance?: number, seamAllowance?: number): void => {
+        if (!this._sizeRatio) {
+            throw new Error("the size ratio was not yet defined for this pattern.");
+        }
+
+        this._allowanceSizes = new Map();
+        this._allowanceSizes.set(PatternPathType.Edge, (edgeAllowance || 0.625) * this._sizeRatio);
+        this._allowanceSizes.set(PatternPathType.Fold, 0);
+        this._allowanceSizes.set(PatternPathType.Seam, (seamAllowance || 0.625) * this._sizeRatio);
+    };
+
     // Precondition: arePatternPiecesEnclosed returned true 
     findPatternPieces = (): void => {
         const faces = FaceFinder.FindFaces(this._patternPaths);
-        this._patternPieces = faces.map(face => face.map(i => this._patternPaths[i]));
         
-        // Logging the pattern pieces for debugging
-        this._patternPieces.forEach(patternPiece => {
-            console.log("pattern piece: ");
-            patternPiece.forEach(patternPath => {
-                console.log("patternPath: type " + patternPath.getType().toString() + ", length " + patternPath.getLengthInPixels());
-            });
-        });
+        this._patternPieces = faces.map(face => (
+            new PatternPiece(face.map((edge) => {
+                if (edge.isReversed) {
+                    return this._patternPaths[edge.index].reversedClone();
+                } else {
+                    return this._patternPaths[edge.index].clone();
+                }
+            }))
+        ));
+
+        // Temporary step to inspect pattern pieces and allowances on final review page while developping.
+        this._patternPaths = this._spreadPatternPieces(this._patternPieces);
```

Comment:
I think if we leave the _spreadPatternPieces method implementation, we might as well keep the  line of code calling it. Maybe I could just comment it out? both the line and the method.

Reply:
But calling here does modify this._patternPaths. Calling spread moves the allowance out from the corresponding pattern path right? Why is the new vector x 50 * I?

Reply:
No, that's not what spread does. It slides the i'th PatternPiece 50*i pixels to the right, so that we can examine them a bit spread out from each other. I we didn't spread them at all, the  paths that appear in 2 different patternpieces would print exactly one over the other and it would be hard to see if the pieces look ok.

The reason I modify this_patternpaths is because that's what the renderer catches and displays and I didn't want to change the renderer to catch allowances because we decided not to display them to the user. So it's a hack.

Reply:
You will be welcome to delete all of _spreadPatternPieces and the temporary this._patternPath assignment line once we can examine the results on the pdf, but I thought in the meantime, for developing purposes, we should have a way to see what all of this code does.

Reply:
Ok, thanks for explaining. we can leave it in here for now.